qcfractal/interface/models/torsiondrive.py

```diff
@@ -223,6 +225,61 @@ def get_final_molecules(self, key: Union[int, Tuple[int, ...], str]=None) -> Dic
 
         return self._organize_return(data, key)
 
+
+    def get_final_results(self, key: Union[int, Tuple[int, ...], str]=None) -> Dict[str, Any]:
+        """Returns the final opt gradient result records at each grid point
+
+        Parameters
+        ----------
+        key : Union[int, str, None], optional
+            Specifies a single entry to pull from.
+
+        Returns
+        -------
+        final_results : Dict[str, qcfractal.interface.models.records.ResultRecord]
+            Returns ResultRecord at each grid point in a dictionary or at a
+            single point if a key is specified.
+
+        Examples
+        --------
+        >>> mols = torsiondrive_obj.get_final_results()
+        >>> type(mols[(-90, )])
+        qcfractal.interface.models.records.ResultRecord
+
+        >>> type(torsiondrive_obj.get_final_results((-90,)))
+        qcfractal.interface.models.records.ResultRecord
+
+        """
+
+        if "final_results" not in self.cache:
+
+            map_id_key = {}
+            ret = {}
+            for k, tasks in self.get_history().items():
+                k = self._serialize_key(k)
+                minpos = self.minimum_positions[k]
+                final_opt_task = tasks[minpos]
+                if len(final_opt_task.trajectory) > 0:
+                    final_grad_record_id = final_opt_task.trajectory[-1]
+                    # store the id -> grid id mapping
+                    map_id_key[final_grad_record_id] = k
+            # combine the ids into one query
+            query_result_ids = list(map_id_key.keys())
+            # put the ids into batch of 10k limit
+            batch_size_limit = 10000
+            for i_batch in range((len(query_result_ids)+batch_size_limit-1) // batch_size_limit):
```

Comment:
These will never hit the batch size and batch sizes will be handled differently in the future. My previous batch size comments was mostly as an FYI that this may happen in other circumstances. Can you simplify the code based on the assumption that you will not hit the batch size?

Reply:
OK sure.